app/utils/AuthProvider.tsx

```diff
@@ -0,0 +1,118 @@
+'use client';
+
+import {
+  createContext,
+  ReactNode,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import { useRouter } from 'next/navigation';
+import supabase from '../../api/supabase/createClient';
+
+// Define the shape of the context
+interface AuthContextType {
+  authUser: any;
```

Comment:
@ronniebeggs curious if you have thoughts on whether storing the `AuthUser` object vs. the `Session` in the AuthContext would be better? i did [the latter in ijp](https://github.com/calblueprint/immigration-justice-project/blob/c2cc9a3d250bdcf9859376106ab3db8e1ee6fea2/src/utils/AuthProvider.tsx#L17) but tbh i have no idea why